py/mattak/backends/uproot/dataset.py

```diff
@@ -278,8 +349,18 @@ def wfs(self, calibrated : bool = False) -> numpy.ndarray:
             if len(wf_idxs): 
                 w[wf_idxs] = self._wfs['radiant_data[24][2048]'].array(entry_start=wf_start, entry_stop=wf_end, library='np')
 
-        # here we'd eventually handle calibration I think? 
-        if self.multiple: 
+        # calibration
+        if calibrated and self.cal_file:
+            cal_param = self.__unpack_cal_parameters(self.cal_file)
```

Comment:
Probably it would be better to not call those in this function right? This faction can be called several times per dataset while the `__unpack` functions only need to be called once per dataset